cpp_exercises/day3/phoneBook/phonebook.cpp

```diff
@@ -16,3 +17,12 @@ const Contact* PhoneBook::find(const std::string& name) {
     }
     return nullptr;
 }
+
+std::ostream& operator<<(std::ostream& ostream, const PhoneBook& phoneBook) {
+    for(size_t i = 0; i < phoneBook.m_contacts.size(); i++) {
+        const Contact& contact = phoneBook.m_contacts[i];
+        ostream << i + 1 << ") " << contact.surname << ' ' << contact.name << '\n';
+        ostream << "   tel: " << contact.telephoneNumber << '\n';
+    }
```

Comment:
Puoi usare `std::for_each()` tipo così (non ricordo di preciso cosa prenda in input la lambda...):
```
    std::for_each(begin(phoneBook.m_contacts), end(phoneBook.m_contacts), [ostream&](const auto &c) {
  ostream << c.surname ...;
});
```
Vero è che questa soluzione toglie il numero del contatto. Purtroppo non mi pare esista niente di semplice in C++ per fare la enumerate di un container. Forse in C++20 hanno aggiunto qualcosa con i ranges, ma non mi sono ancora aggiornato su questa parte.